### ISSUE TITLE
Guard:Nanoc fails to start with nanoc 4.11.1

### ISSUE THREAD

**Issue by Fjan** [nanoc/nanoc#1409]
### Steps to reproduce

1. Any nanoc repo with a guard file and guard-nanoc gem, with nanoc 4.11.1
2. start Guard: `bundle exec guard`

### Actual behavior

Guard::Nanoc crashes. Downgrading nanoc to 4.11.0 it works fine again.

### Crash log

```
11:30:40 - ERROR - Guard::Nanoc failed to achieve its <start>, exception was:
> [#b39debf6872d] NoMethodError: undefined method `<<' for nil:NilClass
> [#b39debf6872d] /Users/jmfaber/.rvm/gems/ruby-2.5.3/gems/nanoc-4.11.1/lib/nanoc/cli/commands/compile_listeners/abstract.rb:54:in `on'
```
full trace here: https://gist.github.com/Fjan/6642e259d5bc9e4afcb5b32eff4716cc

I tried to do a quick comparison between 4.11.0 and 4.11.1 but the changes were quite extensive so I wasn't able to pin down the commit that broke it.

Is the guard-nanoc using some undocumented end point and should the issue be filed there instead?

**Comment by ddfreyne** [comment]
Hey, this is an incompatibility introduced in Nanoc 4.11.1 for which I have a Guard::Nanoc release prepared, but forgot about. My bad! An updated Guard::Nanoc release is coming up.

**Comment by ddfreyne** [comment]
@Fjan This is fixed in the just-released guard-nanoc 2.1.5. Can you confirm?

**Comment by Fjan** [comment]
Hi @ddfreyne 
Guard now starts up but as soon as anything changes and it tries to compile anything it produces an error with a long stack trace: https://gist.github.com/Fjan/6cbbff28125adff6694f2df1484b776a

I tried this on two different nanoc projects.

**Comment by ddfreyne** [comment]
Yikes, I think I screwed up something a bit larger than I expected. Will investigate.

**Comment by ddfreyne** [comment]
I’ve done some more digging and this seems to be caused by the new implementation of `NotificationCenter`, which is now threaded. Guard::Nanoc forks, and forking likely messes up the threads enough for the notification center to stop working. Nasty…

**Comment by ddfreyne** [comment]
I have a fix in #1410. I have no idea how to test this in an automatic way; I’ve verified that it works with my own sites for now…

**Comment by Fjan** [comment]
Thnx for your quick attention to this! (Out of town this week but will test when I get back)
